### DISCUSSION TITLE
Different Header from page 2

### DISCUSSION THREAD

**Discussion by adaxer** [QuestPDF/QuestPDF#90] [Q&A] [answered]
Hello,

can there be a header starting at page 2 for the rest of the document, which is different from the header on the first page?
I know, there can be 2 sections with different page composition, but if the content of the first section spans eg 2 pages, then the new header would start on page 3.

What I would like to accomplish is, that on the first page there is a certain title and every other page can have a "subtitle" at the top.

Thanks again for building this great library :-)

Regards
Andreas

**Comment by MarcinZiabek** [comment, marked as answer]
Hello 😁 You can use the ShowOnce and SkipOnce components to achieve your requirement. Take a look [here](https://github.com/QuestPDF/QuestPDF/pull/28).

**Reply by adaxer** [reply]
Hello Marcin,

what a great way to do this! Thanks for sharing the link as well. I read the conversation there and it was very inspiring. I also do stumble upon use cases from time to time - if you don't mind I will get back to you if there is something that looks as if it could improve QuestPDF (even more ;-))

Thanks and take care
Andreas

**Reply by MarcinZiabek** [reply]
Sure 😁 Any feedback and suggestions are always welcome. Please don't hesitate to share them!

**Comment by MarcinZiabek** [comment]
I have just updated the documentation to cover this use case. [Link](https://www.questpdf.com/documentation/patterns-and-practices.html#achieving-different-header-footer-on-the-first-page)

**Reply by adaxer** [reply]
I just saw, that this was obviously already documented with SkipOnce ([https://www.questpdf.com/documentation/api-reference.html#skip-once](url)), with exactly this use case (different headers). I don't know how I missed that, I was sure I scanned the whole doc - sorry

**Reply by MarcinZiabek** [reply]
Yes, technically it was already documented 😁 However, it only becomes straightforward, once you know the ShowOnce/SkipOnce trick. Otherwise, you don't know where to look for. I still see value in a separate article. 

This is something that I am slowly learning over time - how to write good and useful documentation. And I already see that QuestPDF may benefit from rewriting it at some point 😀

**Comment by adaxer** [comment]
Yeah, agree – but there are not many projects out there with a documentation as good as QuestPDFs is already.

Von: Marcin Ziąbek ***@***.***>
Gesendet: Donnerstag, 6. Januar 2022 13:07
An: QuestPDF/QuestPDF ***@***.***>
Cc: Andreas Daxer ***@***.***>; Author ***@***.***>
Betreff: Re: [QuestPDF/QuestPDF] Different Header from page 2 (Discussion #90)


Yes, technically it was already documented 😁 However, it only becomes straightforward, once you know the ShowOnce/SkipOnce trick. Otherwise, you don't know where to look for. I still see value in a separate article.

This is something that I am slowly learning over time - how to write good and useful documentation. And I already see that QuestPDF may benefit from rewriting it at some point 😀

—
Reply to this email directly, view it on GitHub<https://github.com/QuestPDF/QuestPDF/discussions/90#discussioncomment-1918107>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ACBDMAI5MKIFTC4UCS3K4ZLUUWAVTANCNFSM5LKRBX4Q>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.

**Reply by MarcinZiabek** [reply]
Thank you, I'm glad that you find the documentation good 😀 

Over time I started to recognize that the general development experience comes not only from library/framework quality but also from the learning experience. I am switching languages and technologies quite often and therefore I am not able to remember most of the stuff. Having good documentation and API reference at hand is key for me!

If you find any potential improvement in this area, please feel free to propose changes or prepare pull request yourself 😁

**Comment by MarcinZiabek** [comment]
Hello 😁 I have just released the QuestPDF 2022.5 version. It introduces the `DynamicComponent` structure that gives more precise control over the content visible on each page. [Please take a look if it solves your requirement.](https://www.questpdf.com/documentation/patterns-and-practices.html#dynamic-components)

**Reply by adaxer** [reply]
Sorry for the delay, the SkipOnce and ShowOnce solved the former issue for me, but this dynamic component is very interesting in other scenarios. I will definitely come across it :-)

**Comment by DawidRobertNowak** [comment]
Hello,

We are closing this thread due to inactivity. Thank you for taking the time to reach out and contribute to the project — your feedback helps us improve QuestPDF.

Should you encounter any further problems, please don't hesitate to open a new issue.

Thank you!
